### PR TITLE
opencv 4.7.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

**Comment by chenrui333** [comment]
`pyqt@5` bottle issue on linux build

**Comment by chenrui333** [comment]
rebased per the `qt` and `qt@5` rev updates.

**Comment by fxcoudert** [comment]
Revision bumps needed: `caffe mlt mxnet siril visp`

File: Formula/caffe.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class Caffe < Formula
   url "https://github.com/BVLC/caffe/archive/1.0.tar.gz"
   sha256 "71d3c9eb8a183150f965a465824d01fe82826c22505f7aa314f700ace03fa77f"
   license "BSD-2-Clause"
-  revision 42
+  revision 43
```
Comment (SMillerDev) [inline Formula/caffe.rb:7]: It really should be deprecated. The releasenotes for 1.0 mention that it's going into maintenance mode but it hasn't seen a commit in over 2 years and the successor has been abandoned in favour of pytorch

Comment (iMichka) [reply Formula/caffe.rb:7]: Done in #119433. Let's rev bump this here though. We will have to rebase anyway as `opencv` did not build anyway.

**Comment by chenrui333** [comment]
mxnet artifact location would be handled in #119057

**Comment by chenrui333** [comment]
rebased after the mxnet update.

**Pull request closed without merging**